PennMobile/src/main/java/com/pennapps/labs/pennmobile/GsrFragment.kt

```diff
@@ -0,0 +1,519 @@
+        val minutes = calendar.get(Calendar.MINUTE)
+        val hour = calendar.get(Calendar.HOUR)
+        val day = calendar.get(Calendar.DAY_OF_MONTH)
+        val month = calendar.get(Calendar.MONTH) + 1
+        val year = calendar.get(Calendar.YEAR)
+        val ampm = calendar.get(Calendar.AM_PM)
+
+        calendarButton?.text = (month.toString() + "/" + day + "/" + year)
+
+        // Set default start/end times for GSR booking
+        val ampmTimes = getStartEndTimes(hour, minutes, ampm)
+        startButton?.text = ampmTimes[0]
+        endButton?.text = ampmTimes[1]
+
+        // Set up recycler view for list of GSR rooms
+        gsrRoomListRecylerView = v.findViewById<View>(R.id.gsr_rooms_list) as RecyclerView
+
+        /**
+         * On Click functions for buttons
+         */
+
+        //set start time button
+        startButton?.setOnClickListener(View.OnClickListener {
+            // Get Current Time
+            val c = Calendar.getInstance()
+            val mHour = c.get(Calendar.HOUR_OF_DAY)
+            val mMinute = c.get(Calendar.MINUTE)
+
+            // Launch Time Picker Dialog
+            val timePickerDialog = TimePickerDialog(activity,
+                    TimePickerDialog.OnTimeSetListener { view, hourOfDay, minute ->
+                        val AM_PM: String
+                        if (hourOfDay < 12) {
+                            AM_PM = "AM"
+                        } else {
+                            AM_PM = "PM"
+                        }
+
+                        var hourString = Integer.toString(hourOfDay)
+                        if (hourOfDay == 0) {
+                            hourString = "12"
+                        }
+                        //convert to nonmilitary time
+                        if (hourOfDay > 12) {
+                            hourString = Integer.toString(hourOfDay - 12)
+                        }
+
+                        var minuteString = Integer.toString(minute)
+
+                        //Android treats minutes less than 10 as single digit
+                        if (minute < 10) {
+                            minuteString = "0$minute"
+                        }
+
+                        //display selected time
+                        startButton?.text = "$hourString:$minuteString $AM_PM"
+                        searchForGSR()
+                    }, mHour, mMinute, false)
+            timePickerDialog.show()
+        })
+
+        //end time button
+        endButton?.setOnClickListener(View.OnClickListener {
+            // Get Current Time
+            val c = Calendar.getInstance()
+            val mHour = c.get(Calendar.HOUR_OF_DAY)
+            val mMinute = c.get(Calendar.MINUTE)
+
+            // Launch Time Picker Dialog
+            val timePickerDialog = TimePickerDialog(activity,
+                    TimePickerDialog.OnTimeSetListener { view, hourOfDay, minute ->
+                        val AM_PM: String
```

Comment:
Ternary operator as well. In fact, start and end buttons have identical click listeners so you could probably create the click listener separately and give the two buttons new instances of the click listener.